modules/generic/src/main/scala/ShowModule.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+
+package schema
+
+package generic
+
+trait ShowModule[R <: Realisation] extends GenericSchemaModule[R] {
+  import Schema._
+
+  implicit val showDecidableInstance: Decidable[Show] = new Decidable[Show] {
+    override def choose2[Z, A1, A2](a1: => Show[A1], a2: => Show[A2])(f: Z => A1 \/ A2): Show[Z] =
+      Show.shows[Z](z => f(z).fold(a1.shows, a2.shows))
+
+    override def conquer[A]: Show[A] = Show.shows[A](_ => "")
+
+    override def divide[A, B, C](fa: Show[A], fb: Show[B])(f: C => (A, B)): Show[C] = Show.shows[C](
+      c => {
+        val tpl = f(c)
+
+        val lhs = fa.shows(tpl._1)
+        val rhs = fb.shows(tpl._2)
+
+        if (lhs != "" && rhs != "") s"""($lhs, $rhs)"""
```

Comment:
Cosmetics: I usually prefer to `match` on `(lhs, rhs)` in this kind of situation.

Reply:
More nitpicking: I suppose it would be more efficient to use `show` and the `Cord#isEmpty` methods here.

Reply:
switched to using `Cord` and `isEmpty`. The matching however then requires guards unless there's a pattern for an empty `Cord` ?